tests/test_worker.py

```diff
@@ -1063,11 +1063,11 @@ def test_clean_queue_registries(self):
         self.assertEqual(self.testconn.zcard(foo_registry.key), 0)
         self.assertEqual(self.testconn.zcard(bar_registry.key), 0)
 
-        # worker.clean_registries() only runs once every 15 minutes
-        # If we add another key, calling clean_registries() should do nothing
+        # worker.clean_registries() releases the maintenance locks after cleaning registries.
+        # If we add another key and call clean_registries, the key should be deleted
         self.testconn.zadd(bar_registry.key, {'bar': 1})
         worker.clean_registries()
-        self.assertEqual(self.testconn.zcard(bar_registry.key), 1)
+        self.assertEqual(self.testconn.zcard(bar_registry.key), 0)
```

Comment:
Since the cleaning lock was added manually, shouldn't the manually added registry lock still be present? The worker shouldn't be able to acquire lock and thus, doesn't attempt to clean (and release the lock).

Reply:
@selwin it looks like the manually added lock's key is rq:wip:bar, not rq:clean_registries:bar.

Reply:
Also I apologize, is there a difference between the cleaning lock and the registry locK?

Reply:
Good question, let me investigate further when I have the time.

Reply:
So `rq:wip:bar` is the Redis key for `bar` queue's `StartedJobRegistry`. 

`self.assertEqual(self.testconn.zcard(bar_registry.key), 0)` means that the registry is empty, meaning `worker.clean_registries()` did indeed run.

Since the decision whether to clean registries is now handled by `should_run_maintenance_tasks`, I think we can delete the last test case.

Also, do you mind changing the assertions from `self.assertEqual(self.testconn.zcard(foo_registry.key), 0)` to `self.assertEqual(len(foo_registry), 0)`?

Reply:
@selwin Done